text/0061-defrag-memos.md

```diff
@@ -0,0 +1,68 @@
+- Feature Name: RTH Defragmentation Memos
+- Start Date: 2023-02-03
+- MCIP PR: [mobilecoinfoundation/mcips#0061](https://github.com/mobilecoinfoundation/mcips/pull/0061)
+
+# Summary
+[summary]: #summary
+
+This proposal introduces a new memo type to support [Recoverable Transaction History (RTH)](https://github.com/mobilecoinfoundation/mcips/pull/4). The new memo type will be used by clients to mark `TxOut`s created as part of account defragmentation. This can be used by client applications as a reliable way to exclude defragmentation transactions from appearing as regular transactions.
+
+# Motivation
+[motivation]: #motivation
+
+[RTH](https://github.com/mobilecoinfoundation/mcips/pull/4) introduced a standard method of utilizing [encrypted memos](https://github.com/mobilecoinfoundation/mcips/pull/3) for recording transaction history between two users. Not all transactions on the MobileCoin blockchain are between two different users or accounts, however. Defragmenation transactions are transactions that a user sends to themselves in order to consolidate small `TxOut`s into larger ones. This is necessary if a user needs to send a large transaction but cannot satisfy the amount using 16 `TxOut`s or less.
+
+From the client's perspective, these larger `TxOut`s created by defragmentation appear as `TxOut`s received through normal transactions. This would be confusing to users if displayed this way in an application. Currently, the only way to differentiate defragmentation transactions from regular ones is to use [RTH](https://github.com/mobilecoinfoundation/mcips/pull/4) and check if the sender and recipient are both the user's account. This isn't immediately clear though, especially to new developers who have not been introduced to concepts like fragmentation or [RTH](https://github.com/mobilecoinfoundation/mcips/pull/4). Thus, a new memo type is proposed to mark defragmentatino transactions.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+This proposal adds one new encrypted memo type, the Defragmentation Memo. Defragmentation transactions will write this memo to the payload `TxOut`. Defragmentation transactions will also not produce a change `TxOut`, as there should never be change and it is not needed to construct RTH. This memo has type bytes `0x0002`.
```

Comment:
So, part of the reason that we currently write a change output, even if it's value is zero, is that it makes more of your transactions indistinguishable.

With this MCIP, if I see you submit a transaction and only one output is produced, I know that you are submitting a self-payment.

It's better not to reveal that if we don't have to. And the cost of writing an additional zero value output to the chain is pretty low. So I'd suggest to still write the change output anyways.